dpe/src/commands/derive_child.rs

```diff
@@ -15,36 +15,122 @@ pub struct DeriveChildCmd {
     handle: ContextHandle,
     data: [u8; DPE_PROFILE.get_hash_size()],
     flags: u32,
-    tcb_type: u32,
+    tci_type: u32,
     target_locality: u32,
 }
 
 impl DeriveChildCmd {
-    const _INTERNAL_INPUT_INFO: u32 = 1 << 31;
-    const _INTERNAL_INPUT_DICE: u32 = 1 << 30;
-    const _RETAIN_PARENT: u32 = 1 << 29;
-    const _TARGET_IS_DEFAULT: u32 = 1 << 28;
+    const INTERNAL_INPUT_INFO: u32 = 1 << 31;
+    const INTERNAL_INPUT_DICE: u32 = 1 << 30;
+    const RETAIN_PARENT: u32 = 1 << 29;
+    const MAKE_DEFAULT: u32 = 1 << 28;
+    const CHANGE_LOCALITY: u32 = 1 << 27;
 
-    const fn _is_internal_input_info(&self) -> bool {
-        self.flags & Self::_INTERNAL_INPUT_INFO != 0
+    const fn uses_internal_info_input(&self) -> bool {
+        self.flags & Self::INTERNAL_INPUT_INFO != 0
     }
 
-    const fn _is_internal_input_dice(&self) -> bool {
-        self.flags & Self::_INTERNAL_INPUT_DICE != 0
+    const fn uses_internal_dice_input(&self) -> bool {
+        self.flags & Self::INTERNAL_INPUT_DICE != 0
     }
 
-    const fn _is_retain_parent(&self) -> bool {
-        self.flags & Self::_RETAIN_PARENT != 0
+    const fn retains_parent(&self) -> bool {
+        self.flags & Self::RETAIN_PARENT != 0
     }
 
-    const fn _is_target_is_default(&self) -> bool {
-        self.flags & Self::_TARGET_IS_DEFAULT != 0
+    const fn makes_default(&self) -> bool {
+        self.flags & Self::MAKE_DEFAULT != 0
+    }
+
+    const fn changes_locality(&self) -> bool {
+        self.flags & Self::CHANGE_LOCALITY != 0
+    }
+
+    /// Check if it will be able to create a new default.
+    ///
+    /// # Arguments
+    ///
+    /// * `parent_idx` - Index of the soon-to-be parent.
+    /// * `default_context_idx` - Index of the current default context, if there is one.
+    fn safe_to_make_default(&self, parent_idx: usize, default_context_idx: Option<usize>) -> bool {
+        if let Some(default_idx) = default_context_idx {
+            if default_idx != parent_idx || self.retains_parent() {
+                return false;
+            }
+        }
+        true
     }
 }
 
 impl<C: Crypto> CommandExecution<C> for DeriveChildCmd {
-    fn execute(&self, _dpe: &mut DpeInstance<C>, _locality: u32) -> Result<Response, DpeErrorCode> {
-        Err(DpeErrorCode::InvalidCommand)
+    fn execute(&self, dpe: &mut DpeInstance<C>, locality: u32) -> Result<Response, DpeErrorCode> {
```

Comment:
Would it make sense to call this 'cmd' instead of 'self'? Idk, to me 'self' invokes connotations of an instance of a thing that's longer-lived than a single command invocation. Could just be me, up to you.

Reply:
That would be a more fitting name probably, but in Rust `self` is a keyword to indicate it is a member function of a variable. Kind of like python.

Reply:
Ah ack.

Reply:
Putting self as the first parameter is the way you denote it is a member function. If you exclude it, it will be a static function.